SYCL/Basic/accessor/accessor.cpp

```diff
@@ -166,6 +166,29 @@ int main() {
     }
   }
 
+  // Device accessor with 2-dimensional subscript operators for atomic accessor
+  // check compile error
+  {
+    cl::sycl::queue queue;
+    if (!queue.is_host()) {
+      cl::sycl::range<2> range(1, 1);
+      int Arr[] = {2};
+      {
+        cl::sycl::buffer<int, 1> Buf(Arr, 1);
+        queue.submit([&](cl::sycl::handler &cgh) {
+          auto acc =
+              cl::sycl::accessor<int, 2, cl::sycl::access::mode::atomic,
+                                 cl::sycl::access::target::local>(range, cgh);
+          cgh.parallel_for<class dim2_subscr_atomic>(
+              range, [=](cl::sycl::id<2>) {
```

Comment:
As far as I get it, local accessor shouldn't work with `parallel_for` accepting `range` in both SYCL-1.2.1 and SYCL-2020.
After implementation becomes conformant the test will fail.
Hence, it needs fixing.

Reply:
I changed parameter of parallel_for. Accordingly to SYCL-2020 local accessor must not be used only for parallel_for with range parameter